modules/api/src/main/java/org/apache/ignite/tx/IgniteTransactions.java

```diff
@@ -125,7 +127,9 @@ default CompletableFuture<Transaction> beginAsync() {
      * }
      * </pre>
      *
-     * <p>If the closure is executed normally (no exceptions) the transaction is automatically committed.
+     * <p>If the closure is executed normally (no exceptions) the transaction is automatically committed. In a case of exception, the
+     * closure will be retried automatically within the transaction timeout, so it must be pure function. If the transaction timeout
```

Comment:
Any transaction should be pure function, meaning that no effect should be observed in case of rollback. Are we going to retry the transaction in case of any exception, e.g. ConstraintViolationException?

Reply:
Or, on the contrary add the list of retryable ones, which seems more robust.

Reply:
done